### PR TITLE
add Checkin Screen with Slide Component as well as update Realm schema and Context methods

### PR DESCRIPTION
**Pull request by Juett**

### CODE REVIEW

File: js/components/ChooseAvatar/Avatar.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  Image
+} from "react-native";
+import LinearGradient from "react-native-linear-gradient";
+import globalStyles from "../../config/styles";
+import styles from "./styles";
+
+export default class Avatar extends Component {
+  constructor(props) {
+    super(props);
+  }
```
Comment (siddharthparmar7) [inline js/components/ChooseAvatar/Avatar.js]: remove constructor. You are not creating any state.

File: js/components/ChooseAvatar/Avatar.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  render() {
+    const { avatar, style } = this.props;
+    return (
+      <View>
```
Comment (siddharthparmar7) [inline js/components/ChooseAvatar/Avatar.js]: use Fragment.

File: js/context/UserContext/UserProvider.js [hunk 3]
```diff
@@ -59,7 +77,12 @@ class UserProvider extends Component {
           addUserToken: this.addUserToken,
           removeUserToken: this.removeUserToken,
           updateUser: this.updateUser,
-          getUser: this.getUser
+          getUser: this.getUser,
+          addActivity: this.addActivity,
+          removeActivity: this.removeActivity,
+          addJournal: this.addJournal,
+          removeJournal: this.removeJournal,
+          firstTime: this.firstTime
```
Comment (siddharthparmar7) [inline js/context/UserContext/UserProvider.js]: firstTime what? Please name your props better.

File: js/context/UserContext/UserProvider.js [hunk 4]
```diff
@@ -30,27 +35,40 @@ class UserProvider extends Component {
...
+    await queryAddJournal(journalID, imageUrl, hashTags);
+    this.getUser();
+  };
+
+  removeJournal = async journalID => {
+    await queryRemoveJournal(journalID);
+    this.getUser();
+  };
+
+  firstTime = async (showOnBoarding, showPrivacyPolicy) => {
```
Comment (siddharthparmar7) [inline js/context/UserContext/UserProvider.js]: name your function better

File: js/context/UserContext/UserProvider.js [hunk 5]
```diff
@@ -3,7 +3,12 @@ import {
   queryAddUserToken,
   queryRemoveUserToken,
   queryUpdateUser,
-  queryGetUser
+  queryGetUser,
+  queryAddActivity,
+  queryRemoveActivity,
+  queryAddJournal,
+  queryRemoveJournal,
+  queryFirstTime
```
Comment (siddharthparmar7) [inline js/context/UserContext/UserProvider.js]: name this better.

File: js/screens/Checkin/Checkin.js [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+                        end={{ x: 0, y: 1 }}
+                        colors={[
+                          globalStyles.blueGradientColor.start.color,
+                          globalStyles.blueGradientColor.end.color
+                        ]}
+                        style={[
+                          StyleSheet.absoluteFill,
+                          {
+                            height: "100%",
+                            width: "100%",
```
Comment (siddharthparmar7) [inline js/screens/Checkin/Checkin.js]: use Dimensions and move your styles to your styles.js file

File: js/screens/Checkin/Checkin.js [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+                    globalStyles.blueGradientColor.start.color,
+                    globalStyles.blueGradientColor.end.color
+                  ]}
+                  style={[
+                    StyleSheet.absoluteFill,
+                    {
+                      height: "100%",
+                      width: "100%",
+                      borderRadius: 20,
+                      padding: 20
```
Comment (siddharthparmar7) [inline js/screens/Checkin/Checkin.js]: same as above.

File: js/screens/Checkin/CheckinContainer.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+export default class CheckinContainer extends Component {
+  static navigationOptions = {
+    headerStyle: {
+      ...globalStyles.navigationHeader
+    }
+  }
+  render() {
+    return (
+      <View>
```
Comment (siddharthparmar7) [inline js/screens/Checkin/CheckinContainer.js]: use Fragment.

File: js/screens/Dashboard/Dashboard.js [hunk 9]
```diff
@@ -1,40 +1,54 @@
...
+                borderRadius: 20,
+                padding: 20
+              }
+            ]}
+          />
+          <Text style={styles.checkin}>Checkin</Text>
+        </TouchableOpacity>
       </View>
     </ImageBackground>
+    // <Checkin navigation={navigation} />
```
Comment (siddharthparmar7) [inline js/screens/Dashboard/Dashboard.js]: remove comments

File: js/screens/Dashboard/DashboardContainer.js [hunk 10]
```diff
@@ -2,12 +2,11 @@ import React, { Component } from "react";
 import Dashboard from "./Dashboard";
 import { View } from "react-native";
 
-
 export default class DashboardContainer extends Component {
   render() {
     return (
       <View>
```
Comment (siddharthparmar7) [inline js/screens/Dashboard/DashboardContainer.js]: use fragment.

File: js/screens/Dashboard/Dashboard.js [hunk 11]
```diff
@@ -1,40 +1,54 @@
...
+              globalStyles.pinkGradientColor.start.color,
+              globalStyles.pinkGradientColor.end.color
+            ]}
+            style={[
+              StyleSheet.absoluteFill,
+              {
+                height: "100%",
+                width: "100%",
+                borderRadius: 20,
+                padding: 20
```
Comment (siddharthparmar7) [inline js/screens/Dashboard/Dashboard.js]: move your styles to styles.js file and use Dimensions

File: js/screens/Dashboard/Dashboard.js [hunk 12]
```diff
@@ -1,40 +1,54 @@
...
-  backgroundGradientFrom: "#1E2923",
-  backgroundGradientTo: "#08130D",
-  color: (opacity = 1) => `rgba(26, 255, 146, ${opacity})`
-};
-const Dashboard = () => {
+const Dashboard = ({ navigation }) => {
   return (
     <ImageBackground
       source={require("../../assets/Background/background-1.png")}
       style={{ width: "100%", height: "100%" }}
```
Comment (siddharthparmar7) [inline js/screens/Dashboard/Dashboard.js]: move it to styles.js file

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make the requested changes. Thank you.

File: js/components/ChooseAvatar/Avatar.js (hunk 1 above)
Comment (Juett) [reply js/components/ChooseAvatar/Avatar.js]: done

File: js/components/ChooseAvatar/Avatar.js (hunk 2 above)
Comment (Juett) [reply js/components/ChooseAvatar/Avatar.js]: done

File: js/context/UserContext/UserProvider.js (hunk 3 above)
Comment (Juett) [reply js/context/UserContext/UserProvider.js]: changed to 2 different methods named toggleShowOnBoarding and toggleShowPrivacyPolicy

File: js/context/UserContext/UserProvider.js (hunk 4 above)
Comment (Juett) [reply js/context/UserContext/UserProvider.js]: changed to 2 different functions named toggleShowOnBoarding and toggleShowPrivacyPolicy

File: js/context/UserContext/UserProvider.js (hunk 5 above)
Comment (Juett) [reply js/context/UserContext/UserProvider.js]: Changed queryFirstTime to 2 different queries named queryToggleShowOnBoarding and queryToggleShowPrivacyPolicy. Changed queryAddJournal and queryRemoveJournal to  queryAddJournalEntry and  queryAddJournalEntry

File: js/screens/Checkin/Checkin.js (hunk 6 above)
Comment (Juett) [reply js/screens/Checkin/Checkin.js]: I'm using dimensions on the parent so I'm using percentage on the gradient to fill the parent. I moved all the styles to styles.js.

File: js/screens/Checkin/Checkin.js (hunk 7 above)
Comment (Juett) [reply js/screens/Checkin/Checkin.js]: I'm using dimensions on the parent so I'm using percentage on the gradient to fill the parent. I moved all the styles to styles.js.

File: js/screens/Checkin/CheckinContainer.js (hunk 8 above)
Comment (Juett) [reply js/screens/Checkin/CheckinContainer.js]: done

File: js/screens/Dashboard/Dashboard.js (hunk 9 above)
Comment (Juett) [reply js/screens/Dashboard/Dashboard.js]: done

File: js/screens/Dashboard/DashboardContainer.js (hunk 10 above)
Comment (Juett) [reply js/screens/Dashboard/DashboardContainer.js]: done

File: js/screens/Dashboard/Dashboard.js (hunk 11 above)
Comment (Juett) [reply js/screens/Dashboard/Dashboard.js]: I'm using dimensions on the parent so I'm using percentage on the gradient to fill the parent. I moved all the styles to styles.js.

File: js/screens/Dashboard/Dashboard.js (hunk 12 above)
Comment (Juett) [reply js/screens/Dashboard/Dashboard.js]: done

File: js/components/ChooseAvatar/Avatar.js (hunk 1 above)
Comment (siddharthparmar7) [reply js/components/ChooseAvatar/Avatar.js]: It's not done.

File: js/components/ChooseAvatar/Avatar.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+import {
+  Text,
+  View,
+  Modal,
+  TouchableOpacity,
+  StyleSheet,
+  Image
+} from "react-native";
+import LinearGradient from "react-native-linear-gradient";
+import globalStyles from "../../config/styles";
```
Comment (siddharthparmar7) [inline js/components/ChooseAvatar/Avatar.js]: remove all the unnecessary imports.

File: js/components/ChooseAvatar/Avatar.js (hunk 2 above)
Comment (siddharthparmar7) [reply js/components/ChooseAvatar/Avatar.js]: I don't see it.

File: js/components/ChooseAvatar/ChooseAvatar.js [hunk 14]
```diff
@@ -1,4 +1,4 @@
-import React, { Component, Fragment } from "react";
+import React, { Component } from "react";
 import PropTypes from "prop-types";
```
Comment (siddharthparmar7) [inline js/components/ChooseAvatar/ChooseAvatar.js:2]: remove this import

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
I don't see the changes I asked for. can you make sure that you pushed it? Thank you.

File: js/components/ChooseAvatar/Avatar.js (hunk 1 above)
Comment (Juett) [reply js/components/ChooseAvatar/Avatar.js]: sorry I didnt push before you checked again. they are there

File: js/components/ChooseAvatar/Avatar.js (hunk 2 above)
Comment (Juett) [reply js/components/ChooseAvatar/Avatar.js]: sorry I didnt push before you checked again. they are there

File: js/components/ChooseAvatar/Avatar.js (hunk 13 above)
Comment (Juett) [reply js/components/ChooseAvatar/Avatar.js]: done

File: js/components/ChooseAvatar/ChooseAvatar.js (hunk 14 above)
Comment (Juett) [reply js/components/ChooseAvatar/ChooseAvatar.js:2]: done

**Review by siddharthparmar7 — APPROVED** [review]
Thank you!

**Pull request merged**